sdk/core/Azure.Core/src/Internal/RehydrationOperation.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    internal class RehydrationOperation : Operation
+    {
+        private readonly NextLinkOperationImplementation _nextLinkOperation;
+        private readonly OperationInternal _operation;
+
+        public RehydrationOperation(HttpPipeline pipeline, RehydrationToken? rehydrationToken, ClientOptions options = null)
+        {
+            AssertNotNull(pipeline, nameof(pipeline));
+            AssertNotNull(rehydrationToken, nameof(rehydrationToken));
```

Comment:
Should the nullable annotation be removed from the `rehydrationToken` if it can't be null?

Reply:
Since we return `RehydrationToken?` to the user in [Operation.GetRehdyrationToken](https://github.com/Azure/azure-sdk-for-net/blob/7a1e1f14d5d4c0f7089bb0f700c987a92d155224/sdk/core/Azure.Core/src/Operation.cs#L22), think it's better to align the signature to avoid extra cast.

Reply:
removed nullable to avoid the confusion

Reply:
Removed nullable